cli/flox-activations/src/gen_rc/bash.rs

```diff
@@ -0,0 +1,198 @@
+use std::io::Write;
+use std::path::PathBuf;
+
+use anyhow::Result;
+use shell_gen::{GenerateShell, SetVar, Shell, Source, UnsetVar};
+
+use crate::env_diff::EnvDiff;
+
+/// Arguments for generating bash startup commands
+#[derive(Debug, Clone)]
+pub struct BashStartupArgs {
+    pub flox_activate_tracelevel: u32,
+    pub activate_d: PathBuf,
+    pub flox_env: PathBuf,
+    pub flox_env_cache: Option<PathBuf>,
+    pub flox_env_project: Option<PathBuf>,
+    pub flox_env_description: Option<String>,
+    pub is_in_place: bool,
+    pub clean_up: Option<PathBuf>,
+
+    // Some(_) if it exists, None otherwise
+    pub bashrc_path: Option<PathBuf>,
+    pub flox_sourcing_rc: bool,
+    pub flox_activations: PathBuf,
+}
+
+pub fn generate_bash_startup_commands(
+    args: &BashStartupArgs,
+    env_diff: &EnvDiff,
+    writer: &mut impl Write,
+) -> Result<()> {
+    let mut stmts = vec![];
+
+    // Enable trace mode if requested
+    if args.flox_activate_tracelevel >= 2 {
+        stmts.push("set -x".to_stmt());
```

Comment:
style nonblocking: from first reading through this, it seems like the abstraction is making things more complex, but I'm not really sure what it's gaining us

What advantage is there to:
str -> SetVar -> impl Statement -> Statement::generate(writer) -> SetVar::generate(writer) -> String written
As opposed to:
str -> export_no_expansion() -> String -> write String

Reply:
(still nonblocking)

> That's nice because it removes two arguments from every call site, which I think is valuable because it reduces noise
```
        let var = SetVar::exported_no_expansion("_flox_sourcing_rc", "true");
        stmts.push(var.to_stmt());
```
could be
```
    let var = shell.export_no_expansion("_flox_sourcing_rc", "true");
    lines.push(var);
```

> doesn't require you to allocate a new string just to be written to a bigger string (the current method writes each statement directly to the writer)

If there is a performance difference between putting a Statement in a Vec and a String in a Vec, I don't think it matters